src/dune_engine/fs_memo.ml

```diff
@@ -13,7 +13,8 @@ module Watcher : sig
      directly if there is no parent. This is an optimisation that allows us to
      reduce the number of watched paths: typically, the number of directories is
      a lot smaller than the number of files. *)
-  val watch : try_to_watch_via_parent:bool -> Path.t -> unit Memo.t
+  val watch :
+    try_to_watch_via_parent:bool -> Path.Outside_build_dir.t -> unit Memo.t
 
   (* Invalidate a path after receiving an event from the file watcher. *)
   val invalidate : Path.t -> Memo.Invalidation.t
```

Comment:
Why not push the new type further and turn everything into `Path.Outside_build_dir` in the implementation too? Or is that going to be too invasive for some reason?

Reply:
I did this part of the API as well. There's still a bunch more places where we can tighten the types, but perhaps that should be subsequent work.